webpack.config.js

```diff
@@ -71,23 +72,22 @@ module.exports = {
       CONFIG: config,
       NODE_ENV: JSON.stringify(process.env.NODE_ENV),
     }),
+    new ForkTsCheckerWebpackPlugin(),
```

Comment:
If we're using `@babel/preset-typescript`, then do we need the TypeScript type checking to be run as part of webpack? Although the project setup in sheltertech.org is different, due to it being built off the Gatsby framework, the way we have it set up there is that Gatsby itself is using `@babel/plugin-transform-typescript`, which just strips away the types, and we just have a separate [npm script to run `tsc` by itself](https://github.com/ShelterTechSF/sheltertech.org/blob/be5eb9fa689ea228c59f21e7f405a70e3b42d816/package.json#L14). You can run it with `npm run -- tsc --watch` to have it run a daemon that watches for file changes.

I think keeping it separate would make the Webpack setup simpler and have fewer moving parts, since the TypeScript side of it would be entirely standalone. Also, for people using IDEs or language servers, they don't even need Webpack running TypeScript since their editor will be doing it itself. I think `fork-ts-checker-webpack-plugin` makes a bit more sense to use if we were using `ts-loader`, since then we would already be depending on the TypeScript compiler within Webpack, but in the setup you have here, `@babel/plugin-transform-typescript` removes the need to have the TypeScript compiler in the loop at all.

Reply:
@babel/preset-typescript also doesn't check types, just strips them away, so by itself we would rely only on the developer using a IDE to check types for them. Not convinced that helps much

ForkTsCheckerWebpackPlugin already runs in the background in a separate process, and is non blocking, so it makes sense to me to keep otherwise it would be perfectly easy to develop valid JS, push and merge a PR, that would not compile as typescript.

Reply:
> @babel/preset-typescript also doesn't check types, just strips them away, so by itself we would rely only on the developer using a IDE to check types for them. Not convinced that helps much

Right, I meant to suggest that we directly run `tsc` in CI to type check the project so that it's still strictly enforced. I think I still prefer that over bundling it into the webpack process, since you get flexibility of running them separately or simultaneously. I may be overly cautious about throwing more stuff into webpack because I've been feeling a lot of pain of trying to do upgrades of libraries when there are a bunch of interdependencies that have to be bumped at the same time.

That said, I'm not adamant against adding this, since the actual integration itself is just adding a single line to the webpack plugins list, without requiring any extra configuration, so if you feel that this is important, then it's fine with me.

Reply:
Will leave for now then, if it causes bloat or notable slowdowns in contributors development, can easily remove at a later date.